cmd/kueuectl/app/list/list_pods.go

```diff
@@ -0,0 +1,245 @@
+	k8s "k8s.io/client-go/kubernetes"
+
+	"sigs.k8s.io/kueue/client-go/clientset/versioned/scheme"
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+)
+
+const (
+	podLong = `Lists all pods that matches the given criteria: Should be part of the specified Job kind,
+	belonging to the specified namespace, matching
+	the label selector or the field selector.`
+	podExample = `  # List Pods
+  kueuectl list pods --for job/job-name`
+)
+
+type PodOptions struct {
+	PrintFlags *genericclioptions.PrintFlags
+
+	AllNamespaces          bool
+	Namespace              string
+	LabelSelector          string
+	FieldSelector          string
+	UserSpecifiedForObject string
+	ForObject              objectRef
+
+	Clientset k8s.Interface
+
+	PrintObj printers.ResourcePrinterFunc
+
+	genericiooptions.IOStreams
+}
+
+func NewPodOptions(streams genericiooptions.IOStreams) *PodOptions {
+	return &PodOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("").WithTypeSetter(scheme.Scheme),
+		IOStreams:  streams,
+	}
+}
+
+func NewPodCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewPodOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "pods --for [<type>[.<api-group>]/]<name>",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"po"},
+		Short:                 "List Pods belong to a Job Kind",
+		Long:                  podLong,
+		Example:               podExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	o.PrintFlags.AddFlags(cmd)
+
+	addAllNamespacesFlagVar(cmd, &o.AllNamespaces)
+	addFieldSelectorFlagVar(cmd, &o.FieldSelector)
+	addLabelSelectorFlagVar(cmd, &o.LabelSelector)
+	addForObjectFilterFlagVar(cmd, &o.UserSpecifiedForObject)
+
+	return cmd
+}
+
+// Complete takes the command arguments and infers any remaining options.
+func (o *PodOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.Namespace, _, err = clientGetter.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.Clientset, err = clientGetter.K8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	if !o.PrintFlags.OutputFlagSpecified() {
+		o.PrintObj = printPodTable
+	} else {
+		printer, err := o.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+		o.PrintObj = printer.PrintObj
+	}
+
+	o.ForObject, err = parseForObjectFilterFlag(o.UserSpecifiedForObject)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run prints the pods for a specific Job
+func (o *PodOptions) Run(ctx context.Context) error {
+	err := o.listPods(ctx)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// listPods lists the pods based on the given --for object
+func (o *PodOptions) listPods(ctx context.Context) error {
+	namespace := o.Namespace
+	if o.AllNamespaces {
+		namespace = ""
+	}
+
+	continueToken := ""
+	filteredPodListSlice := make([]*corev1.PodList, 0)
```

Comment:
Maybe it's better to track totalCount instead slice to do not take memory? Look [here](https://github.com/kubernetes-sigs/kueue/blob/4c9502c02f07af8515d40754ff121b995d44c0bf/cmd/kueuectl/app/list/list_localqueue.go#L135).

Reply:
Thanks! Done https://github.com/kubernetes-sigs/kueue/pull/2280/commits/b5a12206958932fcdbedcac55aaeeb16a7b19b3a